store/store.go

```diff
@@ -39,6 +39,11 @@ type ProjectJobSpecRepository interface {
 	GetJobNamespaces(ctx context.Context) (map[string][]string, error)
 }
 
+// InterProjectJobSpecRepository represents a storage interface for Job specification
+type InterProjectJobSpecRepository interface {
+	GetWithFilters(ctx context.Context, projectName, jobName, resourceDestination string) ([]models.JobSpec, error)
```

Comment:
Instead of having one function to accepts all the parameters, we suggest to have it like this:
* we can split the function into two, like `GetByJobName` with return `[]models.JobSpec` and `GetByDestination` with return `models.JobSpec` as it is unique and expected to return only one (of course all of them with error as well in the return)
* we can put the responsibilities of selecting which repository to be called to service layer. such case can be:
  * get by project name and job name, we can use `ProjectJobSpecRepository.GetByName`
  * get by project name, we can use `ProjectJobSpecRepository.GetAll`
  * get by job name (only) is already mentioned in the first point
  * get by destination (only) is already mentioned in the first point
  * get by project name and destination, we don't need it as destination is unique
  * get by job name and destination, we don't need it as destination is unique
  * get by project name, job name, and destination, we don't need it as destination is unique
  * if none are set (meaning empty), then the service should return error (ensuring at least one being set)

cc: @arinda-arif

Reply:
So request with project name and destination will be served as empty jobsspecs as its not needed?